### ISSUE TITLE
Missing Test Case - 238. Product of Array Except Self

### ISSUE THREAD

**Issue by gitsrj** [LeetCode-Feedback/LeetCode-Feedback#24619]
### LeetCode Username

sahiljaiswal166

### Problem Number, Title, and Link

238. Product of Array Except Self https://leetcode.com/problems/product-of-array-except-self

### Bug Category

Missing test case *(Incorrect/Inefficient Code getting accepted because of missing test cases)*

### Bug Description

According to the problem it should get executed in O(n) time complexity but when the helper function for encountering zero is called it is still completing in O(n). There has to be a test case where the array size is significantly large and consisting of zeroes, that will eventually lead to n square time complexity and heading to a TLE without the solution being submitted.

### Language Used for Code

C++

### Code used for Submit/Run operation

```bash
class Solution {
public:

    int getMult(vector<int>&nums, int index)
    {
        int n= nums.size();
        int ans=1;
        for(int i =0; i<n; i++)
        {
            if(i == index)
            {
                continue;
            }
            else{
                ans *= nums[i];
            }
        }
        return ans;
    }

    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int>ans;

        int allMult = 1;
        for(int i=0; i<n; i++)
        {
            allMult *= nums[i];
        }
        for(int i=0; i<n; i++)
        {
            if(nums[i]!=0)
            {
                int mult = allMult/nums[i];
                ans.push_back(mult);
            }
            else
            {
                ans.push_back(getMult(nums, i));
            }
        }
        return ans;
    }
};
```

### Expected behavior

So basically it should have given a Time Limit Exceeded Error but it did not. It passed all the test cases and the final answer was approved. It should have a test case consisting of an array of large size with almost every element equal to zero, then it will definitely take n square time complexity.
